app-config.yaml

```diff
@@ -3,12 +3,19 @@ app:
   baseUrl: http://localhost:3000
 
 backend:
-  baseUrl: http://localhost:7000
+  baseUrl: https://localhost:7000
   listen: 0.0.0.0:7000
   cors:
     origin: http://localhost:3000
     methods: [GET, POST, PUT, DELETE]
     credentials: true
+  certificate:
```

Comment:
I don't think certificate generation is useful enough to add to the tooling tbh. It can be solved with a simple script that generates certificates in the rare occasion that you actually need to serve with https locally or with a self-signed certificate in a deployment.

What I think we should have here is a `https:` object with ca/cert/key props, and some other things from https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options. Similar to https://webpack.js.org/configuration/dev-server/#devserverhttps. The way you'd load PEM data from a file would be via secrets, e.g.

```yaml
backend:
  https:
    cert:
      $secret:
        file: cert.pem
    key:
      $secret:
        file: key.pem
```

Reply:
I think certificate generation is super useful, because it saves us the trouble of having to generate a cert if we do not have one. I realize this is easy if your on a OS that supports openssl pretty much out of the box but for some people this is not the case and even if it was they do not find this process to be trivial.

Furthermore you can always add in configuration logic to support both scenarios. But granted that I have the reverse view on this I do not feel inclined to change the code. If you do not want it we will simply keep it in our own fork and leave it at that. :)

Reply:
You don't really want to keep regenerating the certificates if you can avoid it, so I think it's better as a separate script anyway. You can use the same `selfsigned` package in a script in your own repo if you want to.